src/measurements/Calibration_Classes.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar  5 13:47:39 2025
+
+@author: katiekoch
+"""
+
+''' Calibration Classes'''
+
+import time
+from PyQt5 import QtCore
+import numpy as np
+from pathlib import Path
+import sys
+from ctypes import *
+import h5py
+import scipy.signal as signal
+from PyQt5.QtWidgets import QApplication, QFileDialog
+import csv
+path_root = Path(__file__).parents[2]
+sys.path.append(str(path_root))
+from src.compute.beams import Beam
+from src.compute.calibration import Calibration
+
+from src.drivers.Slm_Meadowlark_optics import SLM
+from src.drivers.Slm_Meadowlark_optics import ImageGen
+
+class Measure_LUT_PhasetoGreyscale(QtCore.QThread):
+
+    ''' 
+        Runs a measurement that will scan through the greyscale values on half of the SLM display, 
+        while keeping the other half set to zero and records the intensity on a spectrometer.
+    '''
+    sendSpectrum = QtCore.pyqtSignal(np.ndarray, np.ndarray)
+    sendProgress = QtCore.pyqtSignal(float)
+    sendParameter = QtCore.pyqtSignal(str, float)
+
+    def __init__(self, devices, parameter, int_time, spectra_number, scan_number):
+        '''
+         Initializes the LUT file measurement
+         input:
+             - devices: the devices dictionary holding at least a spectrometer and a SLM
+             - parameters: 
+        ''' 
+        
+        super(Measure_LUT_PhasetoGreyscale, self).__init__()
+
+        self.spectrometer = devices['spectrometer']
+        self.SLM= devices['SLM']
+        self.ImageGen = devices['ImageGen']
+        self.int_time = int_time
+        self.spectra_number = spectra_number
+        self.scan_number = scan_number
+        self.GreyScale_Vals = np.arange(0,256,1) #255
+        #print(self.GreyScale_Vals)
+        #print(len(self.GreyScale_Vals))
+        #self.GreyScale_Vals = np.arange(0, 11, 1)  # for testing purposes
+        self.spectra = []  # preallocate spec array
+        self.summedspec = []
+        self.wls = self.spectrometer.get_wavelength()
+        self.terminate = False
+        self.acquire_measurement = True
+
+        self.parameter = parameter
+
+    def run(self):
+        print(time.strftime('%H:%M:%S') + ' Run LUT File Calibration Measurement')
```

Comment:
Could you replace the `print` functions with calls to log if they are still relevant? See the logging section in `manual/contributing.md` . Same goes for other prints in the code.